DuckDuckGo/TabViewController.swift

```diff
@@ -343,14 +349,34 @@ class TabViewController: UIViewController {
         }
     }
     
+    private func getCleanUrl(_ url: URL, completion: @escaping (URL) -> Void) {
+        // Rewrite tracking links
+        if let cleanUrl = linkCleaner.extractCanonicalFromAmpLink(initiator: nil, destination: url) {
+            completion(cleanUrl)
+        } else if ampExtractor.urlContainsAmpKeyword(url) {
+            ampExtractor.getCanonicalUrl(initiator: nil, url: url) { canonical in
```

Comment:
UX of waiting for this to finish is poor, especially on first load - I'd add `showProgressIndicator()` above this line (and in the code below that calls this) to give some visual indicator that things are happening.

Reply:
And maybe add comment in both places that we do this to improve UX.